src/simulation/dynamics/linearTranslationalBodies/linearTranslationBodiesNDOF/_UnitTest/test_linearTranslationNDOFStateEffector.py

```diff
@@ -0,0 +1,806 @@
+
+    # Plotting
+    plt.close("all")
+    plt.figure()
+    plt.clf()
+    plt.plot(timeSec, (orbAngMom_N[:, 0] - initialOrbAngMom_N[0]) / initialOrbAngMom_N[0],
+             timeSec, (orbAngMom_N[:, 1] - initialOrbAngMom_N[1]) / initialOrbAngMom_N[1],
+             timeSec, (orbAngMom_N[:, 2] - initialOrbAngMom_N[2]) / initialOrbAngMom_N[2])
+    plt.xlabel('time (s)')
+    plt.ylabel('Relative Difference')
+    plt.title('Orbital Angular Momentum')
+
+    plt.figure()
+    plt.clf()
+    plt.plot(timeSec, (orbEnergy - initialOrbEnergy) / initialOrbEnergy)
+    plt.xlabel('time (s)')
+    plt.ylabel('Relative Difference')
+    plt.title('Orbital Energy')
+
+    plt.figure()
+    plt.clf()
+    plt.plot(timeSec, (rotAngMom_N[:, 0] - initialRotAngMom_N[0]) / initialRotAngMom_N[0],
+             timeSec, (rotAngMom_N[:, 1] - initialRotAngMom_N[1]) / initialRotAngMom_N[1],
+             timeSec, (rotAngMom_N[:, 2] - initialRotAngMom_N[2]) / initialRotAngMom_N[2])
+    plt.xlabel('time (s)')
+    plt.ylabel('Relative Difference')
+    plt.title('Rotational Angular Momentum')
+
+    plt.figure()
+    plt.clf()
+    plt.plot(timeSec, (rotEnergy - initialRotEnergy) / initialRotEnergy)
+    plt.xlabel('time (s)')
+    plt.ylabel('Relative Difference')
+    plt.title('Rotational Energy')
+
+    plt.figure()
+    plt.clf()
+    plt.plot(rho1Data.times() * 1e-9, rho1, label=r'$\rho_1$')
+    plt.plot(rho2Data.times() * 1e-9, rho2, label=r'$\rho_2$')
+    plt.plot(rho3Data.times() * 1e-9, rho3, label=r'$\rho_3$')
+    plt.plot(rho4Data.times() * 1e-9, rho4, label=r'$\rho_4$')
+    plt.legend(loc='best')
+    plt.xlabel('time (s)')
+    plt.ylabel('Displacement')
+
+    plt.figure()
+    plt.clf()
+    plt.plot(rho1Data.times() * 1e-9, rho1Dot, label=r'$\dot{\rho}_1$')
+    plt.plot(rho2Data.times() * 1e-9, rho2Dot, label=r'$\dot{\rho}_2$')
+    plt.plot(rho3Data.times() * 1e-9, rho3Dot, label=r'$\dot{\rho}_3$')
+    plt.plot(rho4Data.times() * 1e-9, rho4Dot, label=r'$\dot{\rho}_4$')
+    plt.legend(loc='best')
+    plt.xlabel('time (s)')
+    plt.ylabel('Displacement Rate')
+
+    if show_plots:
+        plt.show()
+    plt.close("all")
+
+    # Testing setup
+    accuracy = 1e-12
+
+    np.testing.assert_allclose(finalOrbEnergy, initialOrbEnergy, rtol=accuracy,
+                               err_msg="Orbital energy is not constant.")
+    np.testing.assert_allclose(finalRotEnergy, initialRotEnergy, rtol=accuracy,
+                               err_msg="Rotational energy is not constant.")
+    for i in range(3):
+        np.testing.assert_allclose(finalOrbAngMom, initialOrbAngMom_N, rtol=accuracy,
+                                   err_msg="Orbital angular momentum is not constant.")
+        np.testing.assert_allclose(finalRotAngMom, initialRotAngMom_N, rtol=accuracy,
+                                   err_msg="Rotational angular momentum is not constant.")
+
+
+def translatingBodyCommandedForce(show_plots):
+    __tracebackhide__ = True
```

Comment:
each method should have a doc-string to explain what this function does. This can be brief as you have a master function above.

Reply:
Done